libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -1032,6 +1032,83 @@ string YulUtilFunctions::suffixedVariableNameList(string const& _baseName, size_
 	return result;
 }
 
+std::string YulUtilFunctions::decrementCheckedFunction(Type const& _type)
+{
+	IntegerType const& type = dynamic_cast<IntegerType const&>(_type);
+
+	string const functionName = "decrement_" + _type.identifier();
+
+	return m_functionCollector->createFunction(functionName, [&]() {
+		u256 minintval;
+
+		// Smallest admissible value to decrement
+		if (type.isSigned())
+			minintval = 0 - (u256(1) << (type.numBits() - 1)) + 1;
+		else
+			minintval = 1;
+
+		return Whiskers(R"(
+			function <functionName>(value) -> ret {
+				if <lt>(value, <minval>) { revert(0,0) }
+				ret := sub(value, 1)
+			}
+		)")
+			("functionName", functionName)
+			("minval", toCompactHexWithPrefix(minintval))
+			("lt", type.isSigned() ? "slt" : "lt")
+			.render();
+	});
+}
+
+std::string YulUtilFunctions::incrementCheckedFunction(Type const& _type)
+{
+	IntegerType const& type = dynamic_cast<IntegerType const&>(_type);
+
+	string const functionName = "increment_" + _type.identifier();
+
+	return m_functionCollector->createFunction(functionName, [&]() {
+		u256 maxintval;
+
+		// Biggest admissible value to increment
+		if (type.isSigned())
+			maxintval = (u256(1) << (type.numBits() - 1)) - 2;
+		else
+			maxintval = (u256(1) << type.numBits()) - 2;
+
+		return Whiskers(R"(
+			function <functionName>(value) -> ret {
+				if <gt>(value, <maxval>) { revert(0,0) }
+				ret := add(value, 1)
+			}
+		)")
+			("functionName", functionName)
+			("maxval", toCompactHexWithPrefix(maxintval))
+			("gt", type.isSigned() ? "sgt" : "gt")
+			.render();
+	});
+}
+
+string YulUtilFunctions::negateNumberCheckedFunction(Type const& _type)
+{
+	IntegerType const& type = dynamic_cast<IntegerType const&>(_type);
+
```

Comment:
Please assert that type is signed.

Reply:
I am already doing that at call site, but I'll move it into the function then